### ISSUE TITLE
CKR_TOKEN_NOT_PRESENT returned for unsupported token

### ISSUE THREAD

**Issue by stsp** [OpenSC/OpenSC#2030]
### Problem Description
Firefox is very slow when token is inserted.
After lots of debugging in this thread:
https://bugzilla.mozilla.org/show_bug.cgi?id=1627273
the conclusion was, based on this log:
https://bug1627273.bmoattachments.org/attachment.cgi?id=9147021
that opensc wrongly returns  CKR_TOKEN_NOT_PRESENT
instead of CKR_TOKEN_NOT_RECOGNIZED.

### Proposed Resolution
Return CKR_TOKEN_NOT_RECOGNIZED for
unsupported token.

### Steps to reproduce
Insert unsupported token.
Run firefox with opensc as a security device driver.

### Logs
https://bug1627273.bmoattachments.org/attachment.cgi?id=9147021

**Comment by Jakuje** [comment]
Thank you for the bug report. It should be pretty easy to fix. Can you check if the PR #2037 solves your issue? 

I tried testing with my card with the following commands in my local build of OpenSC (first recognizes the card, second is selecting driver that is not recognizing the inserted card):
```
PKCS11SPY=./src/pkcs11/.libs/opensc-pkcs11.so ./src/tools/pkcs11-tool -L --module ./src/pkcs11/.libs/pkcs11-spy.so 
OPENSC_DRIVER=cac PKCS11SPY=./src/pkcs11/.libs/opensc-pkcs11.so ./src/tools/pkcs11-tool -L --module ./src/pkcs11/.libs/pkcs11-spy.so
```
 * With card: `C_GetTokenInfo` returns `CKR_OK`
 * Without card the `C_GetTokenInfo` is not called
 * With unrecognized card: `C_GetTokenInfo` returns `CKR_TOKEN_NOT_RECOGNIZED`

**Comment by stsp** [comment]
[logfile.new.txt](https://github.com/OpenSC/OpenSC/files/4652623/logfile.new.txt)

Thanks, the log looks right now.
Not that the amount of transaction
have changed, I think firefox doesn't
properly check the return value.
I'll escalate the problem to firefox
once you merge the fix.

**Comment by Jakuje** [comment]
AFAIK Firefox runs more threads so in this merged output it might look wrong. If you turn on opensc debug log, you will get also some thread information, which might clarify the output.

**Comment by stsp** [comment]
[debug.txt.gz](https://github.com/OpenSC/OpenSC/files/4652681/debug.txt.gz)
Here is the debug output for around 20 seconds
of firefox run.

**Comment by stsp** [comment]
But what exactly in an initial
log looks wrong? I think return
value was right, so what is the
concern?

**Comment by frankmorgner** [comment]
closed with #2030
